enterprise/server/usage/usage.go

```diff
@@ -271,72 +273,65 @@ func (ut *tracker) flushCounts(ctx context.Context, groupID string, c collection
 		Region:          ut.region,
 	}
 	dbh := ut.env.GetDBHandle()
-	return dbh.Transaction(ctx, func(tx *db.DB) error {
+	return dbh.TransactionWithOptions(ctx, db.Opts().WithQueryName("upsert_usage"), func(tx *db.DB) error {
+		tu := &tables.Usage{
+			GroupID:         pk.GroupID,
+			PeriodStartUsec: pk.PeriodStartUsec,
+			Region:          pk.Region,
+			FinalBeforeUsec: c.End().UnixMicro(),
+			UsageCounts:     *counts,
+		}
 		// Create a row for the corresponding usage period if one doesn't already
 		// exist.
-		res := tx.Exec(`
-			INSERT `+dbh.InsertIgnoreModifier()+` INTO "Usages" (
-				group_id,
-				period_start_usec,
-				region,
-				final_before_usec,
-				invocations,
-				cas_cache_hits,
-				action_cache_hits,
-				total_download_size_bytes,
-				linux_execution_duration_usec,
-				mac_execution_duration_usec
-			) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
-			`,
-			pk.GroupID,
-			pk.PeriodStartUsec,
-			pk.Region,
-			c.End().UnixMicro(),
-			counts.Invocations,
-			counts.CASCacheHits,
-			counts.ActionCacheHits,
-			counts.TotalDownloadSizeBytes,
-			counts.LinuxExecutionDurationUsec,
-			counts.MacExecutionDurationUsec,
-		)
-		if err := res.Error; err != nil {
-			return err
+		res := tx.Clauses(clause.OnConflict{DoNothing: true}).Create(tu)
+		if res.Error != nil {
+			return res.Error
 		}
-		// If we inserted successfully, no need to update.
-		if res.RowsAffected > 0 {
+		if res.RowsAffected != 0 {
+			// Insert worked; we're done.
 			return nil
 		}
+
 		// Update the usage row, but only if collection period data has not already
 		// been written (for example, if the previous flush failed to delete the
 		// data from Redis).
-		return tx.Exec(`
-			UPDATE "Usages"
-			SET
-				final_before_usec = ?,
-				invocations = invocations + ?,
-				cas_cache_hits = cas_cache_hits + ?,
-				action_cache_hits = action_cache_hits + ?,
-				total_download_size_bytes = total_download_size_bytes + ?,
-				linux_execution_duration_usec = linux_execution_duration_usec + ?,
-				mac_execution_duration_usec = mac_execution_duration_usec + ?
-			WHERE
-				group_id = ?
-				AND period_start_usec = ?
-				AND region = ?
-				AND final_before_usec <= ?
+		return tx.Model(&tables.Usage{}).Where(`
```

Comment:
Was this necessary because the previous query did not work for postgres?

Reply:
That's correct, postgres has a very different syntax for ignoring conflicts.

Reply:
Oh, you're right, I'm sorry. This change did end up not being necessary and this state is a result of iterative development when fixing this transaction as a whole; I can revert this query.